plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiBackgroundSplitLoader.java

```diff
@@ -113,18 +130,26 @@ public void run()
     private void generateSplits(boolean useIndex)
     {
         // Attempt to apply partition pruning using partition stats index
-        Optional<List<String>> effectivePartitionsOpt = useIndex && partitionIndexSupportOpt.isPresent() ?
-                partitionIndexSupportOpt.get().prunePartitions(lazyPartitions.get()) : Optional.empty();
+        Map<String, Partition> partitionMap = getPartitionInfos();
+        List<String> allPartitions = new ArrayList<>(partitionMap.keySet());
 
-        Deque<String> partitionQueue = new ConcurrentLinkedDeque<>(effectivePartitionsOpt.orElse(lazyPartitions.get()));
+        List<String> effectivePartitions = Optional.ofNullable(useIndex && partitionIndexSupportOpt.isPresent()
+                ? partitionIndexSupportOpt.get().prunePartitions(allPartitions).orElse(null)
+                : null).orElse(allPartitions);
+
+        List<HiveHudiPartitionInfo> hiveHudiPartitionInfos = effectivePartitions.stream()
+                .map(partitionName -> HudiUtil.buildHiveHudiPartitionInfo(tableHandle, partitionName, partitionMap.get(partitionName)))
+                .toList();
```

Comment:
nit: incorporate the partition pruning based on partition stats index, `HiveHudiPartitionInfo` transformation, etc., into `getPartitionInfos()` so that there is a single method returning `hiveHudiPartitionInfos` or `partitionQueue` to simplify this method for readability?